src/exception/ArrayDataException.java

```diff
@@ -0,0 +1,7 @@
+package exception;
+
+public class ArrayDataException extends Exception {
```

Comment:
Extending from `Exception` is not really correct. Would be enough to extend from `RuntimeException`